### PR TITLE
feat(install): require -g / --global flag

### PR DESCRIPTION
**Pull request by bartlomieju**
In preparation for upcoming changes to `deno install` in Deno 2.

If `-g` or `--global` flag is not provided a warning will be emitted:
```
⚠️ `deno install` behavior will change in Deno 2. To preserve the current behavior use `-g` or `--global` flag.
```

The same will happen for `deno uninstall` - unless `-g`/`--global` flag is provided
a warning will be emitted.

Towards https://github.com/denoland/deno/issues/23062

### CODE REVIEW

File: cli/args/flags.rs [hunk 1]
```diff
@@ -1897,6 +1899,13 @@ These must be added to the path manually if required.")
           .help("Forcefully overwrite existing installation")
           .action(ArgAction::SetTrue))
       )
+      .arg(
+        Arg::new("global")
+          .long("global")
+          .short('g')
+          .help("Install a package or script as an executable available global")
```
Comment (dsherret) [inline cli/args/flags.rs:1906]: Maybe:

```suggestion
          .help("Install a package or script as a globally available executable")
```

File: cli/tools/installer.rs [hunk 2]
```diff
@@ -183,7 +184,13 @@ pub async fn infer_name_from_url(url: &Url) -> Option<String> {
   Some(stem.to_string())
 }
 
-pub fn uninstall(name: String, root: Option<String>) -> Result<(), AnyError> {
+pub fn uninstall(uninstall_flags: UninstallFlags) -> Result<(), AnyError> {
+  let name = uninstall_flags.name;
+  let root = uninstall_flags.root;
+
+  if !uninstall_flags.global {
+    log::warn!("⚠️ `deno install` behavior will change in Deno 2. To preserve the current behavior use `-g` or `--global` flag.");
```
Comment (dsherret) [inline cli/tools/installer.rs:192]:
```suggestion
    log::warn!("⚠️ `deno install` behavior will change in Deno 2. To preserve the current behavior use the `-g` or `--global` flag.");
```

File: cli/tools/installer.rs [hunk 3]
```diff
@@ -241,6 +248,9 @@ pub async fn install_command(
   flags: Flags,
   install_flags: InstallFlags,
 ) -> Result<(), AnyError> {
+  if !install_flags.global {
+    log::warn!("⚠️ `deno install` behavior will change in Deno 2. To preserve the current behavior use `-g` or `--global` flag.");
```
Comment (dsherret) [inline cli/tools/installer.rs:252]:
```suggestion
    log::warn!("⚠️ `deno install` behavior will change in Deno 2. To preserve the current behavior use the `-g` or `--global` flag.");
```

**Review by dsherret — APPROVED** [review]
LGTM

File: tests/integration/install_tests.rs [hunk 4]
```diff
@@ -17,17 +17,22 @@ fn install_basic() {
...
-    .skip_output_check()
-    .assert_exit_code(0);
+    .run();
+
+  output.assert_exit_code(0);
+  let output_text = output.combined_output();
+  assert_contains!(
+    output_text,
+    "`deno install` behavior will change in Deno 2. To preserve the current behavior use `-g` or `--global` flag."
+  );
```
[the remark this answers is not in the document]
Comment (bartlomieju) [reply tests/integration/install_tests.rs:35]: Added

**Pull request merged**